### PR TITLE
EZP-31347: Added functionality to show dropdown above select field

### PR DESCRIPTION
**Pull request by GrabowskiM**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-31347
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

Select list now will be positioned above select field if there is less than 300px to the bottom of screen.

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/bundle/Resources/public/js/scripts/core/custom.dropdown.js [hunk 1]
```diff
@@ -131,10 +133,20 @@
                 return;
             }
 
-            const methodName = this.itemsContainer.classList.contains(CLASS_ITEMS_HIDDEN) ? 'addEventListener' : 'removeEventListener';
+            const isListHidden = this.itemsContainer.classList.contains(CLASS_ITEMS_HIDDEN);
+
+            if (isListHidden) {
+                const viewportHeight = window.innerHeight || document.documentElement.clientHeight;
+                const { top } = this.itemsContainer.getBoundingClientRect();
+
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/core/custom.dropdown.js]:
```suggestion
```

File: src/bundle/Resources/public/js/scripts/core/custom.dropdown.js [hunk 2]
```diff
@@ -131,10 +133,20 @@
...
+                const itemsListMethodName = top + ITEMS_LIST_MAX_HEIGHT > viewportHeight ? 'add' : 'remove';
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/core/custom.dropdown.js]:
```suggestion
                const itemsListMethodName = top + ITEMS_LIST_MAX_HEIGHT > viewportHeight ? 'add' : 'remove';
                
```

File: src/bundle/Resources/public/js/scripts/core/custom.dropdown.js [hunk 3]
```diff
@@ -131,10 +133,20 @@
...
+                this.itemsContainer.classList[itemsListMethodName](CLASS_ITEMS_POSITION_TOP);
+            }
 
             this.itemsContainer.classList.toggle(CLASS_ITEMS_HIDDEN);
-            doc.body[methodName]('click', this.onClickOutside, false);
+
+            const bodyMethodName = isListHidden ? 'addEventListener' : 'removeEventListener';
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/core/custom.dropdown.js]: can be moved before the iff to the other const

File: src/bundle/Resources/public/scss/fieldType/edit/_ezselection.scss [hunk 4]
```diff
@@ -8,6 +8,8 @@
             &__items {
                 border: none;
                 box-shadow: 0 calculateRem(2px) calculateRem(4px) 0 rgba(0, 0, 0, 0.45);
+                max-height: calculateRem(300px);
+                overflow-y: scroll;
```
Comment (dew326) [inline src/bundle/Resources/public/scss/fieldType/edit/_ezselection.scss]: always scroll? I think better will be auto

**Comment by lserwatka** [comment]
Could you merge it up?

**Comment by GrabowskiM** [comment]
Merge up done.

**Pull request merged**